### PR TITLE
Add fixes for SonarCloud Code Smells

### PR DESCRIPTION
**Pull request by kylelaker-ccpo**
This cleans up the vast majority of non-TODO-related code smells in the
code base.

### CODE REVIEW

File: infrastructure/jest.config.js [hunk 1]
```diff
@@ -4,5 +4,6 @@ module.exports = {
   testMatch: ['**/*.test.ts'],
   transform: {
     '^.+\\.tsx?$': 'ts-jest'
-  }
+  },
+  collectCoverage: true,
```
Comment (kylelaker-ccpo) [inline infrastructure/jest.config.js:8]: Note: This starts collecting coverage during an `npm run test` but this _does not_ add SonarCloud coverge checking for the `infrastructure/` directory yet (and that's intentional).

File: packages/api/portfolioDrafts/funding/getFundingStep.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";
 import { ApiSuccessResponse, SuccessStatusCode } from "../../utils/response";
-import { dynamodbDocumentClient as client } from "../../utils/dynamodb";
+import { dynamodbDocumentClient as client } from "../../utils/aws-sdk/dynamodb";
```
Comment (zachclark-ccpo) [inline packages/api/portfolioDrafts/funding/getFundingStep.ts:3]: I am confused what this change actually does?

File: packages/api/utils/response.ts [hunk 3]
```diff
@@ -40,7 +40,7 @@ abstract class Response implements APIGatewayProxyResult {
   body: string;
   headers?: Headers;
   multiValueHeaders?: MultiValueHeaders;
-  isBase64Encoded?: boolean | undefined;
```
Comment (zachclark-ccpo) [inline packages/api/utils/response.ts:43]: nice

File: packages/api/utils/response.ts [hunk 4]
```diff
@@ -40,7 +40,7 @@ abstract class Response implements APIGatewayProxyResult {
...
+  isBase64Encoded?: boolean;
```
Comment (zachclark-ccpo) [inline packages/api/utils/response.ts:43]: good fix here

File: packages/api/portfolioDrafts/createPortfolioDraft.ts [hunk 5]
```diff
@@ -38,7 +38,7 @@ export async function handler(event: APIGatewayProxyEvent): Promise<APIGatewayPr
         Item: item,
       })
     );
-    return new ApiSuccessResponse<PortfolioDraftSummary>(item as PortfolioDraftSummary, SuccessStatusCode.CREATED);
+    return new ApiSuccessResponse<PortfolioDraftSummary>(item, SuccessStatusCode.CREATED);
```
Comment (zachclark-ccpo) [inline packages/api/portfolioDrafts/createPortfolioDraft.ts:41]: Does this not produce an error with Postman?
We are asserting most of our other calls with 
```item as <type>``` in the ApiSuccessResponse

Comment (kylelaker-ccpo) [reply packages/api/portfolioDrafts/createPortfolioDraft.ts:41]: SonarCloud says that this is useless and I agree. We should either specify it in the `<>` _or_ with `as` _or_ neither. But not both. Not sure how Postman would have an issue. I'll provide a URL to test against internally in a moment.

File: packages/api/portfolioDrafts/funding/getFundingStep.ts (hunk 2 above)
Comment (jasonburkert-ccpo) [reply packages/api/portfolioDrafts/funding/getFundingStep.ts:3]: Suppose I could read the docs.  🤷

**Pull request merged**